### PR TITLE
Duplicate signal connections created query loop in Reflectometry Polref GUI

### LINKED ISSUE #14947: Fix Open table query loop in Reflectometry Interface (Polref)
When the current table inside the interface is dirty (edited), opening a new table will result in a pop-up asking you if you're sure you don't want to save the dirty table. When 'No' is clicked this same pop-up will reappear at least 3 times (if you continue to press 'No') before going away. Ideally this query should only happen once

### PR DESCRIPTION
**Pull request by MattKing06**
In the `setTableList` method (QtReflMainView.cpp) each entry in the table list had a connection between the entry being chosen and the `setModel(QString)` method.

The `setModel` method then told the Presenter to open the table from the list and the user was them prompted about unsaved changes. If there were more than 1 item in the table list then each table in the list would have an amount of connections equal to the number of list items as they were not being disconnected.

This PR disconnects those previous signals before reconnecting them again.

**To Test:** 
Before testing make sure you have the data from SampleData-ISIS which can be downloaded from [here](http://sourceforge.net/projects/mantid/files/Sample%20Data/SampleData-ISIS.zip/download)
- Manage your data directories to point to the location of INTER_NR.tbl found inside SampleData-ISIS
- Go to Interfaces -> Reflectometry -> ISIS Reflectometry (Polref)
- Click Reflectometry on the menu bar of the interface
- Click Import .TBL
- Load the INTER_NR.tbl file from SampleData-ISIS and give it a workspace name
- Load the same table as many times as you like, giving it a different workspace name each time
- Edit the table currently in the processing view
- Go to `Reflectometry` on the menu bar and then `Open Table`
- Choose another table to load in
- A prompt will appear telling you that you have unsaved changes, would you like to discard them?
- Press `No`
- Ensure that the pop up does not reappear
- Repeat the edit and open another table step a few times to make sure this behaviour remains constant

**Release Notes:** http://www.mantidproject.org/Release_Notes_3_6_Reflectometry

Fixes #14947

### CODE REVIEW

File: MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp [hunk 1]
```diff
@@ -72,6 +72,7 @@ presenter
 */
 void QtReflMainView::setModel(QString name) {
   m_toOpen = name.toStdString();
+  std::cout << "set model is called";
```
Comment (picatess) [inline MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp]: is this debugging information? if so it should propably be removed.

File: MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp [hunk 2]
```diff
@@ -127,7 +127,16 @@ void QtReflMainView::setTableList(const std::set<std::string> &tables) {
...
+    // we disconnect the signals first so that there are not repeated
+    // connections.
+    // When repeated corrections happen the QMessageBox from openTable()
+    // method in ReflMainViewPresenter will be called multiple times
+    // when 'no' is clicked.
+    disconnect(openTable, SIGNAL(triggered()), m_openMap, SLOT(map()));
+    disconnect(m_openMap, SIGNAL(mapped(QString)), this,
+               SLOT(setModel(QString)));
+    // reconnecting the signals after disconnecting them to ensure
+    // each object has only one of these signals.
```
Comment (picatess) [inline MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp]: Is this the same as using `Qt::UniqueConnection` when you connect signals? See  [here](http://doc.qt.io/qt-4.8/qt.html#ConnectionType-enum).

**Comment by MattKing06** [comment]
Yes! I think that this would be better for the desired behaviour. Is the version of Qt we use > 4.6?

**Comment by picatess** [comment]
Followed test instructions and the pop up appears only once.
Happy to merge

**Pull request merged**